storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+                let mut ddl = String::new();
+                file.read_to_string(&mut ddl).map_storage_err()?;
+
+                Schema::from_ddl(&ddl).map(|schema| schema.column_defs)
+            }
+            false => Ok(None),
+        }?;
+
+        Ok(Some(Schema {
+            table_name: table_name.to_owned(),
+            column_defs,
+            indexes: vec![],
+            created: NaiveDateTime::default(),
+            engine: None,
+        }))
+    }
+
+    fn data_path(&self, table_name: &str) -> PathBuf {
+        let path = self.path_by(table_name, "jsonl");
+
+        PathBuf::from(path)
+    }
+
+    fn schema_path(&self, table_name: &str) -> PathBuf {
+        let path = self.path_by(table_name, "sql");
+
+        PathBuf::from(path)
+    }
+
+    fn path_by(&self, table_name: &str, extension: &str) -> String {
+        let path = format!("{}/{}.{extension}", self.path.display(), table_name);
+
+        path
+    }
+
+    fn scan_data(&self, table_name: &str) -> Result<RowIter> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())?;
+        let data_path = self.data_path(table_name);
+        let lines = read_lines(data_path).map_storage_err()?;
+        let row_iter = lines.enumerate().map(move |(key, line)| -> Result<_> {
+            let hash_map = HashMap::parse_json_object(&line.map_storage_err()?)?;
+            let data_row = match &schema.column_defs {
+                Some(column_defs) => {
+                    let values = column_defs
+                        .iter()
+                        .map(|column_def| -> Result<_> {
+                            let value = hash_map
+                                .get(&column_def.name)
+                                .map_storage_err(JsonlStorageError::ColumnDoesNotExist.to_string())?
+                                .clone();
+                            let data_type = value.get_type();
+                            match data_type {
+                                Some(data_type) => match data_type == column_def.data_type {
+                                    true => Ok(value),
+                                    false => value.cast(&column_def.data_type),
+                                },
+                                None => Ok(value),
+                            }
+                        })
+                        .collect::<Result<Vec<_>>>()?;
+
+                    DataRow::Vec(values)
+                }
+                None => DataRow::Map(hash_map),
+            };
+            let key = Key::I64((key + 1).try_into().map_storage_err()?);
+
+            Ok((key, data_row))
+        });
+
+        Ok(Box::new(row_iter))
+    }
+}
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
+where
+    P: AsRef<Path>,
+{
+    let file = File::open(filename)?;
+    Ok(io::BufReader::new(file).lines())
+}
+
+#[test]
```

Comment:
It's a good idea to put your test code in a module separated by a cfg, so that it only builds in your test environment.

```suggestion
#[cfg(test)]
mod tests {
    #[test]
}
```